src/main/java/frc/robot/subsystems/swerve/ModuleIOTalonFX.java

```diff
@@ -198,12 +202,27 @@ public void updateInputs(ModuleIOInputs inputs) {
     inputs.turnAppliedVolts = turnAppliedVolts.getValueAsDouble();
     inputs.turnCurrentAmps = new double[] {turnCurrent.getValueAsDouble()};
 
-    inputs.odometryDrivePositionsMeters =
-        drivePositionQueue.stream().mapToDouble(Units::rotationsToRadians).toArray();
+    // inputs.odometryDrivePositionsMeters =
+    //     drivePositionQueue.stream().mapToDouble(Units::rotationsToRadians).toArray();
+    // inputs.odometryTurnPositions =
+    //     turnPositionQueue.stream()
+    //         .map(Rotation2d::fromRotations) // should be after offset + gear ratio
+    //         .toArray(Rotation2d[]::new);
+    // drivePositionQueue.clear();
+    // turnPositionQueue.clear();
+
+    inputs.odometryTimestamps =
+        timestampQueue.stream().mapToDouble((Double value) -> value).toArray();
+    inputs.odometryDrivePositionsRad =
+        drivePositionQueue.stream()
+            .mapToDouble(
+                (Double value) -> Units.rotationsToRadians(value) / Module.DRIVE_GEAR_RATIO) // TODO
+            .toArray();
     inputs.odometryTurnPositions =
         turnPositionQueue.stream()
-            .map(Rotation2d::fromRotations) // should be after offset + gear ratio
+            .map((Double value) -> Rotation2d.fromRotations(value / Module.TURN_GEAR_RATIO))
```

Comment:
all of our readings from the modules should be in meters since the motors are configured w a mechanism ratio thingy